tools/distrib/install_all_python_modules.sh

```diff
@@ -13,45 +13,58 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-# TODO: Integrate this into CI to avoid bitrot.
+set -ex
 
-echo "It's recommended that you run this script from a virtual environment."
+BASEDIR=$(dirname "$0")/../..
+PACKAGES="grpcio_channelz  grpcio_csds  grpcio_admin grpcio_health_checking  grpcio_reflection  grpcio_status  grpcio_testing grpcio_csm_observability grpcio_tests"
+
+# Change to grpc repo root
+cd "$BASEDIR";
 
+# unit-tests setup starts from here
 function maybe_run_command () {
-  if python setup.py --help-commands | grep "$1" &>/dev/null; then
-    python setup.py "$1";
+  if python3 setup.py --help-commands | grep "$1" &>/dev/null; then
+    python3 setup.py "$1";
   fi
 }
 
-set -e
+python3 -m pip install --upgrade "cython<4.0.0rc1";
+python3 setup.py install;
 
-BASEDIR=$(dirname "$0")
-BASEDIR=$(realpath "$BASEDIR")/../..
+# Build and install grpcio_tools
+pushd tools/distrib/python/grpcio_tools;
+  ../make_grpcio_tools.py
+  GRPC_PYTHON_BUILD_WITH_CYTHON=1 pip install .
+popd;
 
-PACKAGES="grpcio_channelz  grpcio_csds  grpcio_admin grpcio_health_checking  grpcio_reflection  grpcio_status  grpcio_testing grpcio_csm_observability grpcio_tests"
+# Build and install grpcio_observability
+pushd src/python/grpcio_observability;
+  ./make_grpcio_observability.py
+  GRPC_PYTHON_BUILD_WITH_CYTHON=1 pip install .
+popd;
+
+# Install xds_protos
+pushd tools/distrib/python/xds_protos;
+  GRPC_PYTHON_BUILD_WITH_CYTHON=1 pip install .
+popd;
+
+# Build and install individual gRPC packages
+pushd src/python;
+  for PACKAGE in ${PACKAGES}; do
+    pushd "${PACKAGE}";
+      python3 setup.py clean;
+      maybe_run_command preprocess
+      maybe_run_command build_package_protos
+      python3 -m pip install .;
+    popd;
+  done
+popd;
+
+# Run tests for grpcio_tests
+pushd src/python/grpcio_tests;
```

Comment:
Why are we running tests in the install script? This script should just build/install all packages, correct? If the goal is to use these to ensure that our dev script ran correctly, I would suggest:

- splitting out a top-level script that _invokes_ `install_all_python_modules.sh` and then runs a test after that -- this gives us a separation between our developer tooling and the _test_ for that developer tooling
- let's choose a much smaller subset of tests -- the goal here isn't to test every facet of the gRPC library. I think a test that just imports all relevant modules successfully would be reasonable.

Reply:
1.  Created an uber file containing install and run tests scripts
2. install script is this file contains installation of all modules
3. run tests only runs the tooling tests
4. I think it's ok, to keep all tests in tests script . user can re-run all test by running this script

Reply:
> user can re-run all test by running this script

Users will not be running these tests. Our CI will. The concern is that we are doubling any existing flakiness rate in these tests without a clear testing goal. (we just need to test that installation has succeeded)